extensions/pyRevitTools.extension/checks/radar_check.py

```diff
@@ -0,0 +1,484 @@
+# ________________________________________________________________ MAIN FUNCTION
+# ______________________________________________________________________________
+def check_model_extents(document=doc):
+    unit_system = get_model_units_type(document)
+    # _______________________________________________________________HTML Styles
+    output = script.get_output()
+    output.add_style('cover {color:black; font-size:24pt; font-weight:bold;}')
+    output.add_style('header {color:black; font-size:15pt;}')
+    divider = "_"*100
+    test_score = 0
+    output.print_html(
+     ('<cover>__________:satellite_antenna:__10-Mile Radar___________</cover>'))
+    print(divider)
+    print("")
+    output.print_md('# Checking model placement and coordinates')
+    print(divider)
+    # _____________________________Check the distances of base and survey points
+    baspt, survpt = get_project_base_and_survey_pts(document)
+    basptdistance = abs(calculate_distance(baspt, INTERNAL_ORIGIN))
+    surveydistance = abs(calculate_distance(survpt, INTERNAL_ORIGIN))
+
+    if basptdistance > EXTENT_DISTANCE:
+        output.print_md(BAD_STRG + 
+        'Base Point is more than ' + CRITERIA_STRG)
+    if surveydistance > EXTENT_DISTANCE:
+        output.print_md(BAD_STRG + 
+        'Survey Point is more than ' + CRITERIA_STRG)
+    else:
+        output.print_md(GOOD_STRG + 
+        'Survey Point is less than ' + CRITERIA_STRG)
+    basptdistance = calculate_distance(baspt, INTERNAL_ORIGIN)
+    # ____________________________________Calculate the angle between the points
+    baseptangle = calculate_angle(baspt, INTERNAL_ORIGIN)
+    surveyptangle = calculate_angle(survpt, INTERNAL_ORIGIN)
+    truenorthangle = calculate_angle(baspt, survpt)
+    # _______________________________Print the Project Coordinates and Distances
+    baspt_z = str(convert_values(baspt[2], doc))
+    tbdata = [['Internal Origin Coordinates', 
+        str(INTERNAL_ORIGIN[0]),str(INTERNAL_ORIGIN[1]),str(INTERNAL_ORIGIN[2]),
+        ' ', ' '],
+        ['Project Base Point Coordinates to Internal Origin', 
+            str(convert_values(baspt[0],doc)),
+            str(convert_values(baspt[1],doc)),
+            baspt_z, 
+            str(convert_units(basptdistance, document)), baseptangle],
+        ['Survey Point Coordinates to Internal Origin', 
+            str(convert_values(survpt[0], doc)),
+            str(convert_values(survpt[1], doc)),
+            str(convert_values(survpt[2], doc)),
+            str(convert_units(surveydistance, document)), surveyptangle],
+        ['Project Base Point to Survey Delta X', ' ', ' ', ' ',
+            str(convert_units(calculate_horizontal_distance(
+                baspt, survpt)[1], document))],
+        ['Project Base Point to Survey Delta Y', ' ', ' ', ' ',
+            str(convert_units(calculate_horizontal_distance(
+                baspt, survpt)[2], document))],
```

Comment:
repeated calculation, extract the horizontal distance into a variable (or maybe that function should be 3 functions that return a single value each!)